views/gunDrawing/index.pug

```diff
@@ -18,18 +18,44 @@ block content
     br
 
     h2 Winners
+    //- p= winners
+    //- p= allYears
+
+    div.years
+      each year, index in allYears
+        if index === allYears.length - 1
+          p.year.current
+            = year
+        else
+          p.year
+            = year
+
+  
     for winner in winners
-      .card(id=winner._id)
-        if user
-          .adminControls
-            button(type="button").edit edit
-            button(type="button").delete delete
-        h3.name= winner.fullName
-        h5.location= winner.location
-        p.gun= winner.gun
-        p.date= winner.date.toDateString()
-        .shortLine
+      if winner.date.toDateString().slice(-4) === currentYear
+        .card.current-year(id=winner._id)
+          if user
+            .adminControls
+              button(type="button").edit edit
+              button(type="button").delete delete
+          h3.name= winner.fullName
+          h5.location= winner.location
+          p.gun= winner.gun
+          p.date= winner.date.toDateString()
+          .shortLine
+      else
```

Comment:
There is a lot of duplication here. Having to change the template at all within a card, means one has to change it in both places. Why not have the `if` deciding what class is displayed only decide if it renders like so:

```pug
if winner.date.toDateString().slice(-4) === currentYear
  .card.current-year(id=winner._id)
else 
  .card.hidden(id=winner._id)
```

This way the template is kept smaller, as well as making it easier to edit.

Reply:
This seems to be a limitation of Pug. See [this issue](https://github.com/pugjs/pug/issues/1779) regarding the topic. In the future, using something like Handlebars could be beneficial.